installer/tue-install-impl.bash

```diff
@@ -389,24 +392,19 @@ function tue-install-git
     tue-install-debug "Desired version: $version"
     local _try_branch_res # Will be used in _try_branch_git
     local version_cache_file="$TUE_ENV_DIR/.env/version_cache/$targetdir"
-    if [ -n "$version" ]
-    then
+    for branch in $version $BRANCH
+    do
         mkdir -p "$(dirname "$version_cache_file")"
         echo "$version" > "$version_cache_file"
         _try_branch_res=""
         _try_branch_git "$targetdir" "$version"
-        [ -n "$_try_branch_res" ] && res="${res:+${res} }$_try_branch_res"
-    else
-        rm "$version_cache_file" 2>/dev/null
-    fi
-
-    tue-install-debug "Desired branch: $BRANCH"
-    if [ -n "$BRANCH" ] # Cannot be combined with version-if because this one might not exist
-    then
-        _try_branch_res=""
-        _try_branch_git "$targetdir" "$BRANCH"
-        [ -n "$_try_branch_res" ] && res="${res:+${res} }$_try_branch_res"
-    fi
+        if [ -n "$_try_branch_res" ]
+        then
+            res="${res:+${res} }$_try_branch_res"
+        else
+            rm "$version_cache_file" 2>/dev/null
+        fi
+    done
```

Comment:
This is a change in behaviour to current setup. As in the current setup only the version specified in a target is stored in the cache file, not the version from `--branch`. I don't think we should do that. As I think that these versions should still show up in `tue-status.`

Reply:
Good point.